pkg/util/http_test.go

```diff
@@ -36,6 +36,11 @@ func TestHexStringToTraceID(t *testing.T) {
 			id:       "234567890abcdef", // odd length
 			expected: []byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0x34, 0x56, 0x78, 0x90, 0xab, 0xcd, 0xef},
 		},
+		{
+			id:          "1234567890abcdef ", // trailing space
+			expected:    nil,
+			expectError: true,
```

Comment:
Can you check we get the correct error message and not just an error in general?

Reply:
Thanks for the review.  Good suggestion on checking the error.  I've updated the commit to check the specific errors in the util tests.